sycl/unittests/allowlist/ParseAllowList.cpp

```diff
@@ -39,16 +40,74 @@ TEST(ParseAllowListTests, CheckAllowListMultipleDeviceDesc) {
                                                {{"BackendName", "opencl"},
                                                 {"DeviceType", "acc"},
                                                 {"DeviceVendorId", "0x4321"}}};
-  EXPECT_EQ(ActualValue, ExpectedValue);
+  EXPECT_EQ(ExpectedValue, ActualValue);
 }
 
-TEST(ParseAllowListTests, CheckUnsupportedKeyNameIsHandled) {
+TEST(ParseAllowListTests, CheckUnsupportedKeyNameIsHandledInSingleDeviceDesc) {
   try {
     sycl::detail::AllowListParsedT ActualValue = sycl::detail::parseAllowList(
         "BackendName:level_zero,SomeUnsupportedKey:gpu");
+    throw std::logic_error("sycl::runtime_error didn't throw");
   } catch (sycl::runtime_error const &e) {
-    EXPECT_EQ(e.what(), std::string("Unrecognized key in SYCL_DEVICE_ALLOWLIST "
-                                    "-30 (CL_INVALID_VALUE)"));
+    EXPECT_EQ(std::string("Unrecognized key in SYCL_DEVICE_ALLOWLIST. For "
+                          "details, please refer to "
+                          "https://github.com/intel/llvm/blob/sycl/sycl/doc/"
+                          "EnvironmentVariables.md -30 (CL_INVALID_VALUE)"),
+              e.what());
+  } catch (...) {
+    FAIL() << "Expected sycl::runtime_error";
+  }
+}
+
+TEST(
+    ParseAllowListTests,
+    CheckUnsupportedKeyNameIsHandledInTwoDeviceDescWithFirstContainingRegexValue) {
+  try {
+    sycl::detail::AllowListParsedT ActualValue = sycl::detail::parseAllowList(
+        "DriverVersion:{{value}}|SomeUnsupportedKey:gpu");
+    throw std::logic_error("sycl::runtime_error didn't throw");
+  } catch (sycl::runtime_error const &e) {
+    EXPECT_EQ(std::string("Unrecognized key in SYCL_DEVICE_ALLOWLIST. For "
+                          "details, please refer to "
+                          "https://github.com/intel/llvm/blob/sycl/sycl/doc/"
+                          "EnvironmentVariables.md -30 (CL_INVALID_VALUE)"),
+              e.what());
+  } catch (...) {
+    FAIL() << "Expected sycl::runtime_error";
+  }
+}
+
+TEST(
+    ParseAllowListTests,
+    CheckUnsupportedKeyNameIsHandledInTwoDeviceDescWithFirstContainingFixedValue) {
+  try {
+    sycl::detail::AllowListParsedT ActualValue = sycl::detail::parseAllowList(
+        "BackendName:level_zero|SomeUnsupportedKey:gpu");
+    throw std::logic_error("sycl::runtime_error didn't throw");
+  } catch (sycl::runtime_error const &e) {
+    EXPECT_EQ(std::string("Unrecognized key in SYCL_DEVICE_ALLOWLIST. For "
+                          "details, please refer to "
+                          "https://github.com/intel/llvm/blob/sycl/sycl/doc/"
+                          "EnvironmentVariables.md -30 (CL_INVALID_VALUE)"),
+              e.what());
+  } catch (...) {
+    FAIL() << "Expected sycl::runtime_error";
+  }
+}
+
+TEST(
+    ParseAllowListTests,
+    CheckUnsupportedKeyNameIsHandledInTwoDeviceDescWithFirstAndSecondContainingRegexValue) {
```

Comment:
I would prefer to use a shorter name with `--gtest_filter` option.

Reply:
Done